jupyter_dashboards/nbextension/notebook/dashboard-view/dashboard.js

```diff
@@ -400,10 +422,28 @@ define([
             gc.find('.close-btn').click(function() {
                     self._hideCell(self._getParentCell(this));
                 });
-            gc.find('.add-btn').click(function() {
-                    // show a single cell at full width
+            gc.find('.add-btn-bottom').click(function() {
+                    // show a cell full width on bottom of dashboard
                     self._showCell(self._getParentCell(this), { width: self.opts.numCols });
                 });
+            gc.find('.add-btn-top').click(function() {
+                    // show a cell full width on top of dashboard
+                    self._showCell(self._getParentCell(this), { width: self.opts.numCols }, 0);
+                });
+            gc.find('.add-btn-relative').click(function() {
+                    // show a cell full width relative to notebook position
+                    var $cell = self._getParentCell(this);
+                    var i = $cell.index();
+                    var $prevCell = $cell.parent()
+                        .find('.cell')
+                        .slice(0,i)
+                        .filter('.grid-stack-item')
+                        .last();
```

Comment:
I didn't realize `prev()` existed; it looks like `prevAll('.grid-stack-item').first()` is what I could use here as I need the closest previous non-hidden sibling.